internal/pkg/remote/client.go

```diff
@@ -97,9 +100,12 @@ func NewWithConfig(log *logger.Logger, cfg Config, wrapper wrapperFunc) (*Client
 	}
 
 	hosts := cfg.GetHosts()
-	clients := make([]*requestClient, len(hosts))
-	for i, host := range cfg.GetHosts() {
-		connStr, err := urlutil.MakeURL(string(cfg.Protocol), p, host, 0)
+	hostCount := len(hosts)
+	log.With("hosts", hosts).Debugf(
+		"creating remote client with %d hosts", hostCount, hosts)
```

Comment:
no placeholder for hosts it will be appended in an ugly way, they are also included in a separate field, so maybe we could save some bytes